plugins/askrene/askrene.c

```diff
@@ -532,61 +472,164 @@ void get_constraints(const struct route_query *rq,
 	reserve_sub(rq->reserved, &scidd, max);
 }
 
-struct getroutes_info {
-	struct command *cmd;
-	struct node_id *source, *dest;
-	struct amount_msat *amount, *maxfee;
-	u32 *finalcltv, *maxdelay;
-	const char **layers;
-	struct additional_cost_htable *additional_costs;
-	/* Non-NULL if we are told to use "auto.localchans" */
-	struct layer *local_layer;
-};
+static void json_add_getroutes(
+    struct json_stream *js,
+    struct route **routes,
+    /* FIXME: can't we deduce these amounts from the routes themselves */
+    struct amount_msat *amounts,
```

Comment:
We can (BTW this should be `const struct amount_msat *`) for now, but we're allowed to overpay (and, if we end up with a single route, we should do so).  Similarly we should adjust final_cltv to add shadow routes...

Reply:
Alright. Fixed.